packages/flutter_tools/lib/src/runner/target_devices.dart

```diff
@@ -486,25 +503,32 @@ class TargetDevicesWithExtendedWirelessDeviceDiscovery extends TargetDevices {
       // If there are multiple matches, continue on to wait for all attached
       // and wireless devices to load so the user can select between all
       // connected matches.
-      final List<Device> devices = await _getDeviceById(
+      List<Device> specifiedDevices = await _getDeviceById(
         includeDevicesUnsupportedByProject: includeDevicesUnsupportedByProject,
         includeDisconnected: true,
       );
-      if (devices.length == 1) {
-        Device? matchedDevice = devices.first;
+
+      specifiedDevices = await _removeDevDisabledIOSDevices(specifiedDevices);
+
+      if (specifiedDevices.length == 1) {
```

Comment:
i feel like its better to remove the devices first because if you have 3 matching devices, with two disabled devices, by removing them first you can still return the only one, instead of erroring.

Reply:
I would not expect it to error in that case